[Edit] Python: Matplotlib: .boxplot() (#7300)
Codecademy/docs: content/matplotlib/concepts/pyplot/terms/boxplot/boxplot.md

---
Title: '.boxplot()'
Description: 'Creates box-and-whisker plots to display statistical summaries of datasets.'
Subjects:
  - 'Data Science'
  - 'Data Visualization'
Tags:
  - 'Charts'
  - 'Matplotlib'
  - 'Statistics'
CatalogContent:
  - 'learn-python-3'
  - 'paths/data-science'
---

> The **matplotlib's `.boxplot()`** method is a powerful data visualization function in matplotlib's pyplot module that creates box-and-whisker plots to display the statistical summary of a dataset. This method displays the distribution of data through quartiles, showing the median, first quartile (Q1), third quartile (Q3), and potential outliers in a compact visual format.

Reviewer: ```suggestion
The **matplotlib's `.boxplot()`** method is a powerful data visualization function in matplotlib's [`pyplot`](https://www.codecademy.com/resources/docs/matplotlib/pyplot) module that creates box-and-whisker plots to display the statistical summary of a dataset. This method displays the distribution of data through quartiles, showing the median, first quartile (Q1), third quartile (Q3), and potential outliers in a compact visual format.
```

## Syntax

```pseudo
matplotlib.pyplot.boxplot(x, notch=None, sym=None, vert=None, ...)
```

> > **Note:** TThe ellipses (`...`) indicate that there are many additional optional parameters available—such as `widths`, `patch_artist`, `showmeans`, `boxprops`, and others. These parameters provide detailed control over the style, layout, and display of the boxplot.

Reviewer: ```suggestion
> **Note:** The ellipses (`...`) indicate that there are many additional optional parameters available, such as `widths`, `patch_artist`, `showmeans`, `boxprops`, and others. These parameters provide detailed control over the style, layout, and display of the boxplot.
```

**Parameters:**

> - `x`: The input data (array-like or sequence of arrays). Can be a 1D array for single boxplot or sequence of arrays for multiple boxplots.
> - `notch`: Boolean, optional. If True, creates a notched boxplot to indicate confidence intervals around the median.
> - `sym`: String, optional. Default symbol for outlier points. An empty string hides the outliers.
> - `vert`: Boolean, optional. If True (default), plots boxes vertically. If False, plots horizontally.

Reviewer: ```suggestion
- `x`: The input data (array-like or sequence of arrays). Can be a 1D array for a single boxplot or a sequence of arrays for multiple boxplots.
- `notch`: Boolean, optional. If True, a notched boxplot is created to indicate confidence intervals around the median.
- `sym`: String, optional. Default symbol for outlier points. An empty string hides the outliers.
- `vert`: Boolean, optional. If True (default), plots boxes vertically. If False, plots horizontally.
```

**Return value:**

The method returns a [dictionary](https://www.codecademy.com/resources/docs/python/dictionaries) containing the matplotlib artists used in the boxplot. The dictionary includes keys for 'boxes', 'medians', 'whiskers', 'caps', 'fliers', and 'means'.

## Example 1: Creating a Basic Boxplot using `matplotlib.pyplot.boxplot()`

This example demonstrates how to create a simple boxplot using randomly generated data:

```py
import matplotlib.pyplot as plt
import numpy as np

# Set random seed for reproducibility
np.random.seed(42)

# Generate sample data
data = np.random.normal(100, 15, 200)

# Create the boxplot
plt.figure(figsize=(8, 6))
plt.boxplot(data)
plt.title('Basic Boxplot Example')
plt.ylabel('Values')
plt.show()
```

> The output of this code is:
> 
> ![A simple matplotlib boxplot showing the distribution of normally distributed data with a median line, quartile box, whiskers, and outlier points](https://raw.githubusercontent.com/Codecademy/docs/main/media/boxplot1.png)
> 
> The code generates a dataset with 200 values following a normal distribution with mean 100 and standard deviation 15. The resulting boxplot displays the median as a horizontal line, the box representing the interquartile range (IQR), whiskers extending to the most extreme non-outlier data points, and any outliers as individual points.

Reviewer: ```suggestion
The output of this code is:

![A simple matplotlib boxplot showing the distribution of normally distributed data with a median line, quartile box, whiskers, and outlier points](https://raw.githubusercontent.com/Codecademy/docs/main/media/boxplot1.png)

The code generates a dataset with 200 values following a normal distribution with a mean of 100 and a standard deviation of 15. The resulting boxplot displays the median as a horizontal line, the box representing the interquartile range (IQR), whiskers extending to the most extreme non-outlier data points, and any outliers as individual points.
```

clothing = np.append(clothing, [200, 40])         # Seasonal variations
home_goods = np.append(home_goods, [180])         # Holiday boost
sports = np.append(sports, [150, 30])             # Seasonal impact

# Combine all sales data
sales_data = [electronics, clothing, home_goods, sports]
categories = ['Electronics', 'Clothing', 'Home Goods', 'Sports']

# Create customized boxplot
plt.figure(figsize=(12, 8))
box_plot = plt.boxplot(sales_data,
                      labels=categories,
                      patch_artist=True,  # Fill with colors
                      notch=True,         # Show confidence intervals
                      showmeans=True)     # Show mean values

# Customize colors for each category
colors = ['lightblue', 'lightgreen', 'lightcoral', 'lightyellow']
for patch, color in zip(box_plot['boxes'], colors):
  patch.set_facecolor(color)

# Customize the plot appearance
plt.title('Quarterly Sales Performance Analysis by Product Category',
          fontsize=16, fontweight='bold')
plt.ylabel('Sales (in thousands USD)', fontsize=12)
plt.xlabel('Product Categories', fontsize=12)
plt.grid(axis='y', alpha=0.3)
plt.tight_layout()
plt.show()
```

> The output of this code is:
> 
> ![Colorful customized matplotlib boxplots showing quarterly sales performance across four product categories, with notches and mean indicators](https://raw.githubusercontent.com/Codecademy/docs/main/media/boxplot3.png)
> 
> This example simulates a business scenario where sales data is analyzed across different product categories. The customized boxplot uses colors to distinguish categories, shows confidence intervals through notches, and displays mean values alongside medians. This visualization helps identify which product categories perform best and have the most consistent sales patterns.
> 
> ## Frequently Asked Questions
> 
> ### 1. What is a box plot in Matplotlib?
> 
> A box plot displays data distribution through five statistics: minimum, Q1, median, Q3, and maximum, with outliers shown as individual points.
> 
> ### 2. What is the difference between Seaborn Boxplot and Matplotlib Boxplot?
> 
> Seaborn's boxplot offers better default styling and easier categorical data handling, while Matplotlib's boxplot provides more low-level control and customization options.
> 
> ### 3. How to plot a boxplot in a Python DataFrame?

Reviewer: ```suggestion
The output of this code is:

![Colorful customized matplotlib boxplots showing quarterly sales performance across four product categories, with notches and mean indicators](https://raw.githubusercontent.com/Codecademy/docs/main/media/boxplot3.png)

This example simulates a business scenario where sales data is analyzed across different product categories. The customized boxplot uses colors to distinguish categories, shows confidence intervals through notches, and displays mean values alongside medians. This visualization helps identify which product categories perform best and have the most consistent sales patterns.

## Frequently Asked Questions

### 1. What is a box plot in Matplotlib?

A box plot displays data distribution through five statistics: minimum, Q1, median, Q3, and maximum, with outliers shown as individual points.

### 2. What is the difference between Seaborn Boxplot and Matplotlib Boxplot?

Seaborn's boxplot offers better default styling and easier categorical data handling, while Matplotlib's boxplot provides more low-level control and customization options.

### 3. How to plot a boxplot in a Python `DataFrame`?
```

---
Title: '.boxplot()'
Description: 'Creates box-and-whisker plots to display statistical summary of datasets.'
Subjects:
  - 'Data Science'
  - 'Data Visualization'
Tags:
  - 'Charts'
  - 'Matplotlib'
  - 'Statistics'
CatalogContent:
  - 'learn-python-3'
  - 'paths/data-science'
---

> The **`matplotlib.pyplot.boxplot()`** method is a powerful data visualization function in matplotlib's pyplot module that creates box-and-whisker plots to display the statistical summary of a dataset. This method displays the distribution of data through quartiles, showing the median, first quartile (Q1), third quartile (Q3), and potential outliers in a compact visual format.

Reviewer 2: Include keyword in the first sentence itself. You could write something like -
```suggestion
The matplotlib boxplot method is a powerful data visualization function in matplotlib's pyplot module that creates box-and-whisker plots to display the statistical summary of a dataset. This method displays the distribution of data through quartiles, showing the median, first quartile (Q1), third quartile (Q3), and potential outliers in a compact visual format.
```
